src/codegen/utilities/text_wrappers.py

```diff
@@ -0,0 +1,45 @@
+﻿import textwrap
+
+"""
+ Use global text wrapper objects and reassign properties each time instead
+ of instantiating a new text wrapper object each time, as it is about 2 orders
+ of magnitude faster.
+
+ An extra character is subtracted from the line width of the code text wrapper
+ to allow for closing brackets, since the Mako template will be prettier
+ without lines like [${variable + ']' | wrap(4. 8)}
+"""
+wrapper = textwrap.TextWrapper(width=78, break_long_words=False)
+docstring_wrapper = textwrap.TextWrapper(width=72, break_long_words=False)
+
+
+def wrap(initial_indent, subsequent_indent):
```

Comment:
`subsequent_indent` seems to often be equal to `initial_indent`. You could make it `None` by default. And then add code here that sets it to initial_indent if it is `None`. This could cleanup callers quite a bit!